tests/FSharp.Compiler.ComponentTests/ErrorMessages/Directives.fs

```diff
@@ -0,0 +1,305 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+namespace ErrorMessages
+
+open Xunit
+open FSharp.Test.Compiler
+
+module NoWarn =
+
+    // #nowarn is super forgiving the only real error is FS alerts you that you forgot the error ID
+    [<Fact>]
+    let ``NoWarn Errors F# 8`` () =
+
+        FSharp """
+#nowarn "988"
+#nowarn FS
+#nowarn FSBLAH
+#nowarn ACME 
+#nowarn "FS"
+#nowarn "FSBLAH"
+#nowarn "ACME"
+        """
+        |> withLangVersion80
+        |> asExe
+        |> compile
+        |> shouldFail
+        |> withDiagnostics[
+            (Error 3350, Line 3, Col 9, Line 3, Col 11, "Feature '# directives with non-quoted string arguments' is not available in F# 8.0. Please use language version 'PREVIEW' or greater.")
+            (Error 3350, Line 4, Col 9, Line 4, Col 15, "Feature '# directives with non-quoted string arguments' is not available in F# 8.0. Please use language version 'PREVIEW' or greater.")
+            (Error 3350, Line 5, Col 9, Line 5, Col 13, "Feature '# directives with non-quoted string arguments' is not available in F# 8.0. Please use language version 'PREVIEW' or greater.")
+            (Warning 203, Line 6, Col 1, Line 6, Col 13, "Invalid warning number 'FS'")
+            ]
+
+
+    [<Fact>]
+    let ``NoWarn Errors F# 9`` () =
+
+        FSharp """
+#nowarn FS988
+#nowarn FS
+#nowarn FSBLAH
+#nowarn ACME 
+#nowarn "FS"
+#nowarn "FSBLAH"
+#nowarn "ACME"
+        """
+        |> asExe
+        |> withLangVersionPreview
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Warning 203, Line 3, Col 1, Line 3, Col 11, "Invalid warning number 'FS'")
+            (Warning 203, Line 6, Col 1, Line 6, Col 13, "Invalid warning number 'FS'")
```

Comment:
Why are `FSBLAH` and "FSBLAH", and `ACME` and "ACME" not an error? Conversely, if these aren't errors, why is `FS` and `"FS"` an error?

Just tested it with latest RTM, has this always been an error?

![image](https://github.com/dotnet/fsharp/assets/16015770/6946e090-c572-4f96-95b5-232f8a930806)

and even this:
![image](https://github.com/dotnet/fsharp/assets/16015770/c16d7272-462a-4acb-b8ef-9012e2a65918)

this is so odd...

Reply:
Oh lordy ... because of **stupid bugs** in the existing implementation:
A nowarn that begins with fs must be followed by an integer numeric value. If it begins with anything else then it is ignored..
This is the current shipping product:
![image](https://github.com/dotnet/fsharp/assets/5175830/b845fa30-f8f6-42e3-bd1a-409188de0df0)

I thought about fixing them, but that is a breaking change, because 
- #nowarn "ACME" doesn't fail
- #nowarn "FSBLAH" does fail

I didn't correct the non-quoted version which is possible and non-breaking because I did want the behaviour to be different.

However, if we wanted to, we could "deprecate" the quoted version with a warning and make the errors correct in the non quoted.  Which is actually my preferred solution, but I wanted the fslang process to be as non-controversial as possible.